picard/webservice/__init__.py

```diff
@@ -392,6 +392,9 @@ def _init_timers(self):
     def set_cache(self, cache_size_in_bytes=None):
         if cache_size_in_bytes is None:
             cache_size_in_bytes = CACHE_SIZE_IN_BYTES
+        if cache_size_in_bytes <= 0:
+            log.debug("NetworkDiskCache disabled")
+            return
```

Comment:
As I understand it is not necessarily disabled at this point, just keeps the previous setting. So when I change the cache settings and put it to 0 then it takes a restart to actually disable the cache. Would `manager.setCache`  allow to actually unset it?

Reply:
A restart is needed anyway.

Reply:
I agree that this likely won't be used much.  My suggestion may be to add a note saying something like "Requires restart" to the option page where this is set.  The real difficulty will be notifying (reminding) the user that a restart is required if the setting is changed as part of enabling/disabling an option profile.

Reply:
I left it out on purpose, I think we need something generic for options that requires a restart. We already have such message for certain options.
Having a message listing settings that will only apply after restart would be great, and that's something we can define once per option, so basically when we declare them.

Reply:
@phw: what we could do in case of value change is to only call `QNetworkDiskCache.setMaximumCacheSize()` and unconditionally create the cache directory.

 It would avoid any need to restart as [`QNetworkDiskCache.setMaximumCacheSize()`](https://doc.qt.io/qt-6/qnetworkdiskcache.html#setMaximumCacheSize) "Sets the maximum size of the disk cache to be size.

If the new size is smaller then the current cache size then the cache will call [expire](https://doc.qt.io/qt-6/qnetworkdiskcache.html#expire)()."

Reply:
@zas Yes, that seems reasonable

> I left it out on purpose, I think we need something generic for options that requires a restart.

I totally agree with that.

Reply:
@phw I implemented changing cache size without restart in [839e02f](https://github.com/metabrainz/picard/pull/2395/commits/839e02f15436a7a275598037964a473efc9b4d1d)